src/ape_ethereum/ecosystem.py

```diff
@@ -88,6 +91,34 @@ class Block(BlockAPI):
     """
 
 
+def parse_output_type(output_type: str) -> Union[str, Tuple]:
```

Comment:
If you're unsure about a recursive function, it's usually a good indication to introduce fuzzing into the mix. Perhaps something like:

```py
from hypothesis import strategies as st

PRIMITIVE_TYPES = ["uint256", "address", ...]

@given(complex_type=st.tuples(st.one_of(PRIMITIVE_TYPES), min_length=1, max_length=20))
def test_output_parsing(complex_type):
    assert parse_output_type(",".join(complex_type)) == complex_type
```

Reply:
I added a test; it helped me find an edge case.